### DISCUSSION TITLE
Migrate large application from 1.6R6 to 1.7.14

### DISCUSSION THREAD

**Discussion by drolow** [mozilla/rhino#1364] [Q&A]
Hi Rhino community,

I am currently investigating the migration of my company's Java 11 application from Rhino version 1.6R6 to the latest version, 1.7.14. Our application heavily relies on basic JavaScript without utilizing the latest ECMAScript standards, and we frequently interact between Java and JavaScript components.

Before proceeding with the migration, I would like to inquire about the potential performance benefits of upgrading. Considering our usage of basic JavaScript and the frequent back-and-forth between Java and JavaScript, does it make sense to migrate to Rhino version 1.7.14 in terms of performance improvements?

Additionally, I am facing a specific issue during the migration process. In a piece of JavaScript code where I call Java code, I used to have an object that was correctly invoked as expected in version 1.6R6. However, after upgrading to version 1.7.14, this object seems to be typed as java.lang.reflect.Proxy with an attribute containing an inner class of VMBridge_jdk18. While I can access the methods of the interface my class implements, they all return "null". Has anyone encountered a similar issue, and could you provide guidance on how to adapt my code accordingly ?

Thank you in advance for your time and support.

**Comment by drolow** [comment]
In order to provide more context and details regarding my issue, here is the js code:
`ctx.getOperation('someValue').getObject(ctx, null);`
The first method, getOperation(), successfully returns the Java class. However, the subsequent call to getObject(ctx, null) is where the issue arises. Upon inspecting the code in Java, I found that ctx is a Proxy object, and any method invocation on this proxy returns null, resulting in a NullPointerException because I'm calling a method on this object. It seems that the execution now goes through newInterfaceProxy() in VMBridge_jdk18.

To address this issue, I attempted to use Context.jsToJava() on the proxy, but unfortunately, it did not resolve the problem.

**Comment by rbri** [comment]
regarding performance - i use (a slightly customized) version in HtmlUnit (https://github.com/HtmlUnit/htmlunit). I have not seen any real performance change during the last years (as far as i remember). I guess this will be the same for you.
